jetbrains-core/tst/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtilsTest.kt

```diff
@@ -29,13 +57,187 @@ class EcsExecUtilsTest {
+            .launchType(LaunchType.EC2)
+            .containerInstanceArn("sample-container-123")
+            .overrides(TaskOverride.builder().build())
+            .build()
+        val taskDefinition = TaskDefinition.builder().taskDefinitionArn(taskDefinitionArn).build()
+        val containerInstance = ContainerInstance.builder().containerInstanceArn("sample-container-123").ec2InstanceId("sample-ec2-id").build()
+        val iamInstanceProfile = IamInstanceProfile.builder().arn("sample:instance-profile/IamInstanceProfile123").build()
+        val instanceProfile = InstanceProfile.builder().roles(Role.builder().build()).build()
+        val sampleInstance = Instance.builder().instanceId("sample-ec2-id").iamInstanceProfile(iamInstanceProfile).build()
+        val reservation = Reservation.builder().instances(sampleInstance).build()
+
+        ecsClient.stub {
+            on { describeTasks(any<DescribeTasksRequest>()) } doAnswer { DescribeTasksResponse.builder().tasks(listOf(task)).build() }
+            on { describeContainerInstances(any<DescribeContainerInstancesRequest>()) } doAnswer {
+                DescribeContainerInstancesResponse.builder().containerInstances(containerInstance).build()
+            }
+        }
+        ec2Client.stub {
+            on { describeInstances(any<DescribeInstancesRequest>()) } doAnswer { DescribeInstancesResponse.builder().reservations(reservation).build() }
+        }
+
+        iamClient.stub {
+            on { getInstanceProfile(any<GetInstanceProfileRequest>()) } doAnswer {
+                GetInstanceProfileResponse.builder().instanceProfile(instanceProfile).build()
+            }
+        }
+        resourceCache.addEntry(
+            projectRule.project, EcsResources.describeTaskDefinition(task.taskDefinitionArn()),
+            CompletableFuture.completedFuture(taskDefinition)
+        )
+
+        val taskRoleArnResponse = EcsExecUtils.getTaskRoleArn(projectRule.project, clusterArn, taskArn)
+        assertThat(taskRoleArnResponse).isNull()
+    }
+
+    @Test
+    fun `Ensure that required permissions are present`() {
+        val task = Task.builder()
+            .taskArn(taskArn)
+            .overrides(TaskOverride.builder().build())
+            .build()
+        val evaluationResult = EvaluationResult.builder().evalDecision(PolicyEvaluationDecisionType.ALLOWED).build()
+
+        ecsClient.stub {
+            on { describeTasks(any<DescribeTasksRequest>()) } doAnswer { DescribeTasksResponse.builder().tasks(listOf(task)).build() }
+        }
+
+        iamClient.stub {
+            on { simulatePrincipalPolicy(any<SimulatePrincipalPolicyRequest>()) } doAnswer {
+                SimulatePrincipalPolicyResponse.builder().evaluationResults(evaluationResult).build()
+            }
+        }
+
+        val haveRequiredPermissions = EcsExecUtils.checkRequiredPermissions(projectRule.project, clusterArn, taskArn)
+        assertThat(haveRequiredPermissions).isTrue
```

Comment:
should we test what happens when AWS calls error out too?

Reply:
yes, will add this test!